### ISSUE TITLE
option to de-uglify names (Origin: bugzilla #665506)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#4488]
status NEW severity *enhancement* in component *general* for *---*
Reported in version *1.7.4* on platform *Other*
Assigned to: Dimitri van Heesch

On 2011-12-03 17:18:02 +0000, Jonathan Wakely wrote:

    The GNU C++ standard library, libstdc++, uses doxygen to generate our API
    docs:
    http://gcc.gnu.org/onlinedocs/libstdc++/latest-doxygen/
    
    In order to prevent clashes with user-defined macros, standard library code
    must be "uglified" to use names reserved for the implementation i.e. you
    must use '__var' not 'var'
    
    That uglification is an implementation detail, not part of the API.  It
    would greatly improve our docs if there could be a doxygen option to "de-
    uglify" names in the output.
    
    So given:
    
        /** @param  t  The object to frob.
         */
        template<typename _Tp>
          _Tp frobnicate(_Tp __t);
    
    I'd like the doxygen output to show the function as:
    
        template<typename Tp>
          Tp func(Tp t);
    
    And also to recognise that the @param t relates to the parameter __t (rather
    than warning about undocumented parameters).
    
    The alteration should only be done for function parameters and template
    parameters.  Functions, types and namespaces should not be altered.
    
    Would something like this be possible?  If it would be relatively easy for
    someone unfamiliar with the Doxygen code to implement I could try to do so
    myself.

**Comment by albert-github** [comment]
I don't think it is the task of doxygen to iron out specific "uglification"s imposed by project specific coding rules as in the code the specific `ugly` names are present
It might be possible by using some settings e.g.:
- `PREDEFINED` (has to be done for each "uglified" name though
- using a filter, see e.g. `INPUT_FILTER`

**Comment by jwakely** [comment]
I'll look into INPUT_FILTER, thanks.
